app/service/rest.py

```diff
@@ -1,93 +1,126 @@
+
+
+@service.route('/<service_id>', methods=['GET'])
+def get_service_by_id(service_id):
+    user_id = request.args.get('user_id', None)
+    if user_id:
+        fetched = dao_fetch_service_by_id_and_user(service_id, user_id)
+    else:
+        fetched = dao_fetch_service_by_id(service_id)
+    if not fetched:
+        return jsonify(result="error", message="not found"), 404
+    data, errors = service_schema.dump(fetched)
+    return jsonify(data=data)
+
+
 @service.route('', methods=['POST'])
 def create_service():
-    # TODO what exceptions get passed from schema parsing?
-    service, errors = service_schema.load(request.get_json())
+    data = request.get_json()
+    if not data.get('user_id', None):
+        return jsonify(result="error", message={'user_id': ['Missing data for required field.']}), 400
+
+    user = get_model_users(data['user_id'])
+    if not user:
+        return jsonify(result="error", message={'user_id': ['not found']}), 400
+
+    data.pop('user_id', None)
+    if 'name' in data:
+        data['email_from'] = _email_safe(data.get('name', None))
+
+    valid_service, errors = service_schema.load(request.get_json())
+
     if errors:
         return jsonify(result="error", message=errors), 400
-    # I believe service is already added to the session but just needs a
-    # db.session.commit
-    try:
-        save_model_service(service)
-    except DAOException as e:
-        return jsonify(result="error", message=str(e)), 500
-    return jsonify(data=service_schema.dump(service).data), 201
+
+    dao_create_service(valid_service, user)
+    return jsonify(data=service_schema.dump(valid_service).data), 201
 
 
-@service.route('/<service_id>', methods=['PUT', 'DELETE'])
+def _email_safe(string):
+    return "".join([
+        character.lower() if character.isalnum() or character == "." else ""
+        for character in re.sub("\s+", ".", string.strip())
+    ])
+
+
+@service.route('/<service_id>', methods=['POST'])
 def update_service(service_id):
-    try:
-        service = get_model_services(service_id=service_id)
-    except DataError:
-        return jsonify(result="error", message="Invalid service id"), 400
-    except NoResultFound:
-        return jsonify(result="error", message="Service not found"), 404
-    if request.method == 'DELETE':
-        status_code = 202
-        delete_model_service(service)
-    else:
-        status_code = 200
-        update_dict, errors = service_schema_load_json.load(request.get_json())
-        if errors:
-            return jsonify(result="error", message=errors), 400
-        try:
-            save_model_service(service, update_dict=update_dict)
-        except DAOException as e:
-            return jsonify(result="error", message=str(e)), 500
-    return jsonify(data=service_schema.dump(service).data), status_code
+    fetched_service = dao_fetch_service_by_id(service_id)
+    if not fetched_service:
+        return jsonify(result="error", message="not found"), 404
```

Comment:
This error message is inconsistent with the other 404 messages in the api. Should we create task to update the other messages?